aep/general/0181/aep.md.j2

```diff
@@ -1,5 +1,76 @@
 # Stability levels
 
-**Note:** This AEP has not yet been adopted. See
-[this GitHub issue](https://github.com/aep-dev/aep.dev/issues/9) for more
-information.
+While organizations have their own product life cycles, AEPs refer to the
+_stability_ of an API component using the following terms.
+
+## Alpha
+
+An _alpha_ component undergoes rapid iteration with a known set of users who
+**must** be tolerant of change. The number of users **should** be a
+curated, manageable set, such that it is feasible to communicate with all
+of them individually.
+
+Breaking changes **must** be both allowed and expected in alpha components, and
+users **must** have no expectation of stability.
+
+## Beta
+
+A _beta_ component **must** be considered complete and ready to be declared
+stable, subject to public testing. Beta components **should** be exposed to an
+unknown and potentially large set of users. In other words, beta components
+**should not** be behind an allowlist; instead, they **should** be available to
+the public.
+
+Because users of beta components tend to have a lower tolerance of change, beta
+components **should** be as stable as possible; however, the beta component
+**must** be permitted to change over time. These changes **should** be minimal
+but **may** include backwards-incompatible changes to beta components.
+Backwards-incompatible changes **must** be made only after a reasonable
+deprecation period to provide users with an opportunity to migrate their code.
+This deprecation period **must** be defined at the time of being marked beta.
+
+Beta components **should** be time-boxed and promoted to stable if no issues
```

Comment:
Why isn't this "must" instead of "should". You might change it but the point made earlier seems to imply it MUST be timeboxed. You may change the date as users raise questions/concerns but I don't see going from MUST above to SHOULD here.

Reply:
One reason I could imagine this is a should is to enter a new phase of stability without plans to officially leave it. For example, maybe the product itself is not stable, and you're tying the API lifecycle the application (something I don't think should be done, but I've observed in practice).

The reality is that even if we make this a "must", there's no strict way to enforce this AEP. It's fairly toothless in that regard. If the API didn't adhere to this must, would we call the API itself non-compliant with the AEPs, and therefore invalidated from using any tooling?

OTOH if we do take advantage of API stability levels as part of the spec, it might open some interesting avenues in terms of the tooling we can offer - for example, better guarantees around the types of breaking changes allowed when going from a beta to a GA client.

I can raise this as a discussion  in the weekly. Note that we have a current policy of filing issues to address AIP import issues given the time it takes to address them and we're in pre-ga. So If we do merge, it's not to ignore your point - we'll file an issue so we can keep the discussion going.

Reply:
filed https://github.com/aep-dev/aep.dev/issues/183